pandas/io/parsers.py

```diff
@@ -2195,6 +2257,168 @@ def _maybe_parse_dates(self, values, index, try_parse_dates=True):
+        convertoptions = {
+            "include_columns",
+            "null_values",
+            "true_values",
+            "false_values",
+        }
+        parse_options = {k: v for k, v in kwdscopy.items() if k in parseoptions}
+        convert_options = {k: v for k, v in kwdscopy.items() if k in convertoptions}
+        read_options = pyarrow.ReadOptions(autogenerate_column_names=True)
+        headerexists = True if self.header is not None and self.header >= 0 else False
+        skiprows = self.kwds.get("skiprows")
+        if skiprows is not None:
+            read_options = pyarrow.ReadOptions(skip_rows=skiprows)
+        elif headerexists:
+            read_options = pyarrow.ReadOptions(skip_rows=self.header)
+        table = pyarrow.read_csv(
+            self.src,
+            read_options=read_options,
+            parse_options=pyarrow.ParseOptions(**parse_options),
+            convert_options=pyarrow.ConvertOptions(**convert_options),
+        )
+        frame = table.to_pandas()
+        num_cols = len(frame.columns)
+        if not headerexists:
+            if self.names is None:
+                if self.prefix is not None:
+                    self.names = [f"{self.prefix}{i}" for i in range(num_cols)]
+                elif self.header is None:
+                    self.names = range(num_cols)
+            frame.columns = self.names
+
+        frame = self._date_conversion(
+            frame, self._date_conv, self.parse_dates, keep_date_col=self.keep_date_col
+        )
+
+        if self.index_col is not None:
+            for i, item in enumerate(self.index_col):
+                if is_integer(item):
+                    self.index_col[i] = frame.columns[item]
+            frame.set_index(self.index_col, drop=True, inplace=True)
+
+        if self.kwds.get("dtype") is not None:
+            frame = frame.astype(self.kwds.get("dtype"))
+        return frame
+
+    def _date_conversion(
+        self, data, converter, parse_spec, keep_date_col=False,
+    ):
+
+        orig_names = data.columns
+        columns = list(data.columns)
+
+        date_cols = set()
+
+        if parse_spec is None or isinstance(parse_spec, bool):
+            return data, columns
+
+        if isinstance(parse_spec, list):
+            # list of column lists
+            for colspec in parse_spec:
+                if is_scalar(colspec):
+                    if isinstance(colspec, int) and colspec not in data:
+                        colspec = orig_names[colspec]
+                    data[colspec] = converter(data[colspec].values)
+                else:
+                    new_name, col, old_names = self._try_convert_dates(
+                        converter, colspec, data, orig_names
+                    )
+                    if new_name in data:
+                        raise ValueError(f"New date column already in dict {new_name}")
+                    data[new_name] = col
+                    date_cols.update(old_names)
```

Comment:
This seems to duplicate the existing `_process_date_conversion` quite a bit. Do you see any options to re-use the existing functionality? (I am not sure we want to merge it like this)

I think a main difference is that the existing  `_process_date_conversion` works on a dict of arrays (columns), while here this works on a dataframe? 
It might be that with some slight changes it could be used for both cases?

Reply:
@jorisvandenbossche I tried using the existing function but the frame->col_dict->frame conversion kills performance.(~30% performance decrease(30 ms to 40 ms), even when not parsing dates).